components/datadog/agent/install.go

```diff
@@ -58,6 +58,24 @@ func NewInstaller(vm vm.VM, options ...agentparams.Option) (*Installer, error) {
 		return nil, err
 	}
 
+	lastCommand, updateSystemConfigTrigger, err = updateSystemProbeConfig(
+		vm.GetFileManager(),
+		params.SystemProbeConfig,
+		os,
+		lastCommand)
+	if err != nil {
+		return nil, err
+	}
+
+	lastCommand, updateSecurityAgentConfigTrigger, err = updateSecurityAgentConfig(
```

Comment:
By extracting `params.ExtraAgentConfig` logic  from `updateSystemProbeConfig`, it would be possible to have something like:
```go
for _, input := range struct{ content: string, target: string } {
   { "datadog.yaml",  wholeConfigContent},
   { "system-probe.yaml",  params.SystemProbeConfig},
 { "security-agent.yaml",  params.SecurityAgentConfig} } {
   updateConfig(...)
}

Reply:
Hello Oliver, could you elaborate on this ?

Reply:
Note: We discussed it offline.